app/src/main/java/it/niedermann/owncloud/notes/persistence/NotesDatabase.java

```diff
@@ -225,15 +227,15 @@ private DBNote getNoteFromCursor(long accountId, @NonNull Cursor cursor, boolean
         validateAccountId(accountId);
         Calendar modified = Calendar.getInstance();
         modified.setTimeInMillis(cursor.getLong(4) * 1000);
-
         return new DBNote(
                 cursor.getLong(0),
                 cursor.getLong(1),
                 modified,
                 cursor.getString(3),
                 pruneContent ? "" : cursor.getString(9),
                 cursor.getInt(5) > 0,
-                cursor.getString(6),
+//                cursor.getString(6),
+                getTitleByCategoryId(accountId, cursor.getInt(6)),
```

Comment:
This will slow down the down the app, since you are executing an additional SQL query for every note. Currently, I have about 200 notes and showing them on the main Activity requires a single SQL query. With this change, 201 SQL queries are executed. This is a "no go"!

There are two alternatives to fix this issue:
1. build a "cache" in memory for categories, i.e. load all categories of the current account to a `HashMap` on creation of e.g. `NotesDatabase` and query this cache in `getTitleByCategoryId(...)` (and related methods). But on every change of categories (insertion, deletion, edit), you must update this cache.
2. change the SQL query in `getNotesCustom(...)` and add a JOIN on the categories table. Hint: you can reduce the need for changing code if you change the column names of the categories table: If they differ from column names in the notes table, you may not have to prepend the table name (e.g. `CATEGORIES.TITLE`, because `TITLE` exists in both tables `CATEGORIES` and `NOTES`).

I vote for the second approach. You can remove the method `getTitleByCategoryId(...)`, then.

Reply:
I vote for the second approach, too. But I am thinking about whether renaming the columns of table CATEGORIES is needed. I think that prepending the table name is affordable. What's your opinion?